force-app/main/default/classes/ALLO_Allocations_TDTM.cls

```diff
@@ -1176,4 +1194,56 @@ public class ALLO_Allocations_TDTM extends TDTM_Runnable {
         return (val != null) ? val : defaultVal;
     }
 
+    /**
+     * @description Recreates the Scheduled Payment Allocations for Opportunities whose Allocations were updated as a
+     *              result of A Recurring Donations GAU Allocations being updated.
+     */
+    @testVisible private void recreateScheduledPaymentAllocationsIfNecessary() {
```

Comment:
Was this functionality completely missing before now? If not, I'm curious why it required all this new logic...

Reply:
Yes, this has been an issue going all the way back.